pystiche_papers/sanakoyeu_et_al_2018/_loss.py

```diff
@@ -289,3 +292,122 @@ def transformed_image_loss(
     return ops.FeatureReconstructionOperator(
         transformer_block, score_weight=score_weight
     )
+
+
+class MAEReconstructionOperator(ops.EncodingComparisonOperator):
+    r"""The MAE reconstruction loss is a content loss.
+
+    It measures the mean absolute error (MAE) between the encodings of an
+    ``input_image`` :math:`\hat{I}` and a ``target_image`` :math:`I` :
+
+    .. math::
+
+        \mean |\parentheses{\Phi\of{\hat{I}} - \Phi\of{I}}|
+
+    Here :math:`\Phi\of{\cdot}` denotes the ``encoder``.
+
+    Args:
+        encoder: Encoder :math:`\Phi`.
+        score_weight: Score weight of the operator. Defaults to ``1.0``.
+    """
+
+    def enc_to_repr(self, enc: torch.Tensor) -> torch.Tensor:
+        return enc
+
+    def input_enc_to_repr(
+        self, enc: torch.Tensor, ctx: Optional[torch.Tensor]
+    ) -> torch.Tensor:
+        return self.enc_to_repr(enc)
+
+    def target_enc_to_repr(self, enc: torch.Tensor) -> Tuple[torch.Tensor, None]:
+        return self.enc_to_repr(enc), None
+
+    def calculate_score(
+        self,
+        input_repr: torch.Tensor,
+        target_repr: torch.Tensor,
+        ctx: Optional[torch.Tensor],
+    ) -> torch.Tensor:
+        return torch.mean(torch.abs(input_repr - target_repr))
```

Comment:
Although the implementation is simple, we should use `torch`s [builtin implementation](https://pytorch.org/docs/stable/nn.functional.html#torch.nn.functional.l1_loss) here.